src/boot/efifirmware.c

```diff
@@ -0,0 +1,89 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "efifirmware.h"
+#include "util.h"
+#include <endian.h>
+
+static bool efifw_validate_header(
+                const void *blob,
+                size_t uki_efifw_len,
+                void *ret_fwid_off,
+                void *ret_payload_off) {
+
+        if ((uintptr_t) blob % alignof(EfiFwHeader) != 0)
+                return false;
+
+        const EfiFwHeader *fw_header;
```

Comment:
i'd really move this down to where you assign things, and use sizeof_field() for the sizeof() expression below